From 267ccc053f907b883e90c998e0dd2c5471cb4370 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Thu, 8 Jul 2021 22:09:16 +0000
Subject: [PATCH 07/10] Fix uninitialized buffer allocation in
 file_prepare_stream()

The buffer allocated via gs_alloc_bytes() in file_prepare_stream()
remained uninitialized. Zero entire allocated heap memory,
in order to fix that.
---
 base/sfxcommon.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/sfxcommon.c b/base/sfxcommon.c
index 411404317..1489a4d78 100644
--- a/base/sfxcommon.c
+++ b/base/sfxcommon.c
@@ -204,6 +204,7 @@ file_prepare_stream(const char *fname, uint len, const char *file_access,
         gs_free_object(mem, s, "file_prepare_stream");
         return_error(gs_error_VMerror);
     }
+    memset(buffer, 0, buffer_size);
     if (fname != 0) {
         memcpy(buffer, fname, len);
         buffer[len] = 0;	/* terminate string */
-- 
2.39.1

